em_examples/TDEMHorizontalLoopCylWidget.py

```diff
@@ -0,0 +1,775 @@
+        self.Bz = (Pfz*self.f[src, "b", :]).flatten()
+        self.dBxdt = (-Pfx*self.mesh.edgeCurl*self.f[src, "e", :]).flatten()
+        self.dBzdt = (-Pfz*self.mesh.edgeCurl*self.f[src, "e", :]).flatten()
+
+    def plotField(
+        self, Field='B', view="vec", scale="linear", itime=0, Geometry=True, Scenario=None
+    ):
+        # Printout for null cases
+        if (Field == "B") & (view == "y"):
+            print("Think about the problem geometry. There is NO By in this case.")
+        elif (Field == "dBdt") & (view == "y"):
+            print("Think about the problem geometry. There is NO dBy/dt in this case.")
+        elif (Field == "E") & (view == "x") | (Field == "E") & (view == "z"):
+            print("Think about the problem geometry. There is NO Ex or Ez in this case. Only Ey.")
+        elif (Field == "J") & (view == "x") | (Field == "J") & (view == "z"):
+            print("Think about the problem geometry. There is NO Jx or Jz in this case. Only Jy.")
+        elif (Field == "E") & (view == "vec"):
+            print("Think about the problem geometry. E only has components along y. Vector plot not possible")
+        elif (Field == "J") & (view == "vec"):
+            print("Think about the problem geometry. J only has components along y. Vector plot not possible")
+        elif Field == "Model":
+            fig = plt.figure(figsize=(7, 6))
+            ax = plt.subplot(111)
+            if Scenario == 'Sphere':
+                model2D, mapping2D = self.getCoreModel('Sphere')
+            elif Scenario == 'Layer':
+                model2D, mapping2D = self.getCoreModel('Layer')
+
+            out = self.mesh2D.plotImage(np.log10(mapping2D * model2D), ax=ax)
+            cb = plt.colorbar(
+                out[0], ax=ax, format="$10^{%.1f}$"
+            )
+            cb.set_label("$\sigma$ (S/m)")
+            ax.set_xlabel("Distance (m)")
+            ax.set_ylabel("Depth (m)")
+            ax.set_title("Conductivity Model")
+            plt.show()
+        else:
+            fig = plt.figure(figsize=(10, 6))
+            ax = plt.subplot(111)
+            vec = False
+            if view == "vec":
+                tname = "Vector "
+                title = tname+Field+"-field"
+            elif view == "amp":
+                tname = "|"
+                title = tname+Field+"|-field"
+            else:
+                title = Field + view+"-field"
+
+            if Field == "B":
+                label = "Magnetic field (T)"
+                if view == "vec":
+                    vec = True
+                    val = np.c_[self.Bx, self.Bz]
+                elif view == "x":
+                    val = self.Bx
+                elif view == "z":
+                    val = self.Bz
+                else:
+                    # ax.imshow(self.im)
+                    ax.set_xticks([])
+                    ax.set_yticks([])
+                    return "Dude, think twice ... no By for VMD"
```

Comment:
These won't work: would you mind replacing with a print statement followed by a return statement?